test/xpu_api/random/device_tests/common_for_device_tests.h

```diff
@@ -0,0 +1,123 @@
+// -*- C++ -*-
+//===-- common_for_device_tests.h -----------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+//
+// Abstract:
+//
+// Common functionality for device tests
+
+#ifndef _ONEDPL_RANDOM_DEVICE_TESTS_COMMON
+#define _ONEDPL_RANDOM_DEVICE_TESTS_COMMON
+
+#include <CL/sycl.hpp>
+#include <oneapi/dpl/random>
+#include <limits>
+#include <iostream>
+#include <iomanip>
+
+constexpr auto seed = 777;
+constexpr int N = 96;
+
+auto exception_handler = [] (sycl::exception_list exceptions) {
+    for (std::exception_ptr const& e : exceptions) {
+        try {
+            std::rethrow_exception(e);
+        } catch(sycl::exception const& e) {
+            std::cout << "Caught asynchronous SYCL exception during calculation:\n"
+            << e.what() << std::endl;
+        }
+    }
+};
+
+template <typename Fp>
+int comparison(Fp* r0, Fp* r1, std::uint32_t length) {
+    Fp coeff;
+    int numErrors = 0;
+    for (size_t i = 0; i < length; ++i) {
+        if constexpr (std::is_integral<Fp>::value) {
+            if (((int)r0[i] - (int)r1[i]) > 1 || ((int)r1[i] - (int)r0[i]) > 1) {
+                std::cout << "mismatch in " << i << " element: " << r0[i] << " " << r1[i] << std::endl;
+                ++numErrors;
+            }
+        } else {
+            auto diff = std::fabs(r0[i] - r1[i]);
+            auto norm = std::min((std::fabs(r0[i]) + std::fabs(r1[i])), std::numeric_limits<Fp>::max());
+            if constexpr (std::is_same<Fp, float>::value) {
+                coeff = 5000.0f;
+            } else {
+                coeff = 800000000.0;
+            }
+            if (diff > std::max(std::numeric_limits<Fp>::min(), norm * coeff *
+                        std::numeric_limits<Fp>::epsilon())) {
+                std::cout <<  "mismatch in " << i << " element: "  << std::endl;
+                std::cout << std::setprecision (15) << r0[i]  << std::endl; 
+                std::cout << std::setprecision (15) << r1[i] << std::endl; 
+                ++numErrors;
+            }
+        }
+    }
+    return numErrors;
+}
+
+template<class Distr, class Engine>
+int test(sycl::queue& queue) {
+
+    using result_type = typename Distr::result_type;
+    using scalar_type = typename Distr::scalar_type;
+    
+    Engine engine(seed);
+    Distr distr;
+
+    for (int i = 0; i < 99; i++)
+        distr(engine);

```

Comment:
why do we need this generation?

Reply:
It's done to avoid using the default state in device and host generating which we compare. I think it complicates the test.